src/test/blockchain_tests.cpp

```diff
@@ -0,0 +1,118 @@
+#include <boost/test/unit_test.hpp>
+
+#include "stdlib.h"
+
+#include "rpc/blockchain.cpp"
+#include "test/test_bitcoin.h"
+
+/* Equality between doubles is imprecise. Comparison should be done
+ * with a small threshold of tolerance, rather than exact equality.
+ */
+bool DoubleEquals(double a, double b, double epsilon)
+{
+    return std::abs(a - b) < epsilon;
+}
+
+CBlockIndex CreateBlockIndexWithNbits(uint32_t nbits)
+{
+    CBlockIndex block_index;
+    block_index.nHeight = 46367;
+    block_index.nTime = 1269211443;
+    block_index.nBits = nbits;
+    return block_index;
+}
+
+CChain CreateChainWithNbits(uint32_t nbits)
+{
+    CBlockIndex block_index = CreateBlockIndexWithNbits(nbits);
+    CChain chain;
+    chain.SetTip(&block_index);
```

Comment:
That pointer will most likely point to something unwanted after `return` in the line below.

Reply:
Yep, you're right! That was the cause of the non-deterministic test failures. Please forgive my C++ inexperience. Fixed.